### ISSUE TITLE
-Infinity values are not handled correctly in JSON

### ISSUE THREAD

**Issue by rajsite** [ni/VireoSDK#703]
A user found that when using UnflattenFromJSON for double arrays, if the JSON string encounters -Infinity then:
1. The node errors with code -375011
2. The node returns an array with partially parsed results

See forum: https://forums.ni.com/t5/G-Web-Development-Software/Bug-on-Unflatten-from-JSON/m-p/4361814?lightbox-message-images-4361814=330188i0448810F940767B1#M702

![image](https://github.com/ni/VireoSDK/assets/1588923/68634cf3-7b66-421e-92c3-67dc8e6ff51a)

Didn't thoroughly debug but scanning source the following line looks suspicious which seems to indicate that any negative value will not be treated like Infinity:

https://github.com/ni/VireoSDK/blob/673eb0dea01418f6ca20e6a2a53528a941eef7ca/source/core/TDCodecVia.cpp#L1207-L1207

**Comment by spathiwa** [comment]
Agreed, it looks like that line will cause a -Inf to be treated as if it was an overflow a few lines below and return an error.

**Comment by rajsite** [comment]
Created two AzDo bugs for the issue:
1. [Lack of support for -Infinity](https://dev.azure.com/ni/DevCentral/_workitems/edit/2696525)
2. [Returning partial / incorrect results on error](https://dev.azure.com/ni/DevCentral/_workitems/edit/2696528) (effectively user data corruption on error)
